arangod/GeneralServer/HttpCommTask.cpp

```diff
@@ -150,12 +151,12 @@ int HttpCommTask<T>::on_header_complete(llhttp_t* p) {
       (p->http_major != 1 || p->http_minor != 1)) {
     me->sendSimpleResponse(rest::ResponseCode::HTTP_VERSION_NOT_SUPPORTED,
                            rest::ContentType::UNSET, 1, VPackBuffer<uint8_t>());
-    return HPE_OK;
+    return HPE_USER;
```

Comment:
What is the error code `HPE_USER` used for in arangodb? (what is the significance of this change?)

Reply:
`HPE_USER` is not directly used by arangod, but there are several places that compare the return code of the http processing callbacks with `HPE_OK` (e.g. `if (err != HPE_OK) ...`.
Previously we returned `HPE_OK` from here, but the `sendSimpleResponse` already triggered the sending of a response in the background asynchronously. The `HPE_OK` made us continue reading data from the connection and continuing the parsing. Thus we had a race here, and that could actually lead to a crash. 
Returning `HPE_USER` here means: ok, this is not the same as `HPE_OK`, and it is an error that will abort further processing. But it is not any of the predefined errors provided by llhttp. It is basically a catch-all "custom error" happened (in this case: "unsupported HTTP version" or "request entity too large", for which llhttp doesn't have its own error enum values). The fix was validated with an HTTP fuzzer against 3.9.

Reply:
:+1: